satpy/tests/reader_tests/test_hsaf_nc.py

```diff
@@ -0,0 +1,186 @@
+        [
+            LoadDatasetsParams(FILE_PARAMS[FILE_TYPE_H60], ["rr", "qind"], "mm/h", 3000, "msg_seviri_fes_3km"),
+            LoadDatasetsParams(FILE_PARAMS[FILE_TYPE_H63], ["rr", "qind"], "mm/h", 3000, "msg_seviri_iodc_3km"),
+            LoadDatasetsParams(FILE_PARAMS[FILE_TYPE_H90], ["acc_rr", "qind"], "mm", 3000, "msg_seviri_iodc_3km"),
+        ],
+    )
+    def test_load_datasets(self, params):
+        """Test that datasets can be loaded correctly."""
+        with mock.patch("satpy.readers.hsaf_nc.generic_open") as mock_generic_open, \
+                mock.patch("satpy.readers.hsaf_nc.xr.open_dataset") as mock_open_dataset:
+
+            mock_generic_open.return_value.__enter__.return_value = io.BytesIO(b"fake data")
+            mock_open_dataset.side_effect = fake_hsaf_dataset
+
+            loadables = params.file_type["reader"].select_files_from_pathnames([params.file_type["fake_file"]])
+            params.file_type["reader"].create_filehandlers(loadables)
+
+            datasets = params.file_type["reader"].load(params.loadable_ids)
+            dataset_names = {d["name"] for d in datasets.keys()}
+            assert dataset_names == set(params.loadable_ids)
+
+            # check array shapes and types
+            assert datasets[params.loadable_ids[0]].shape == DEFAULT_SHAPE
+            assert datasets[params.loadable_ids[1]].shape == DEFAULT_SHAPE
+            assert np.issubdtype(datasets[params.loadable_ids[0]].dtype, np.floating)
+            assert np.issubdtype(datasets[params.loadable_ids[1]].dtype, np.integer)
+
+            data = datasets[params.loadable_ids[0]]
+            assert data.attrs["spacecraft_name"] == "Meteosat-8"
+            assert data.attrs["platform_name"] == "Meteosat-8"
+            assert data.attrs["units"] == params.unit
+            assert data.attrs["resolution"] == params.resolution
+            assert data.attrs["start_time"] == dt.datetime(2025, 11, 5, 0, 0)
+            assert data.attrs["end_time"] == dt.datetime(2025, 11, 5, 0, 15)
+            assert data.attrs["area"].area_id == params.area_name
+            assert data.dims == ("y", "x")
+
+
+    @pytest.mark.parametrize(
+        ("file_type", "loadable_ids"),
+        [
+            (FILE_PARAMS[FILE_TYPE_H60], ["rr", "qind"]),
+            (FILE_PARAMS[FILE_TYPE_H63], ["rr", "qind"]),
+            (FILE_PARAMS[FILE_TYPE_H90], ["acc_rr", "qind"]),
+        ],
+    )
+    @pytest.mark.skipif(not os.path.exists(FILE_PARAMS[FILE_TYPE_H60]["real_file"]) or
+                        not os.path.exists(FILE_PARAMS[FILE_TYPE_H63]["real_file"]),
+                                           reason="Real HSAF file not present")
+    def test_real_hsaf_file(self, file_type, loadable_ids):
+        """Test the reader with a real HSAF NetCDF file."""
+        # Select files
+        loadables = file_type["reader"].select_files_from_pathnames([file_type["real_file"]])
+        assert loadables, "No loadables found for the real file"
```

Comment:
when are these real files available? are they easily downloadable from somewhere so that the user can test this?
Otherwise, I think it's as good to create stub files that mimic these files closely.